### PR TITLE
Adds skeleton framework for admin control middleware

### PR DESCRIPTION
**Pull request by bubakazouba**
implements logic for isUpsertCommand

### CODE REVIEW

File: userManager.js [hunk 1]
```diff
@@ -154,6 +154,12 @@ module.exports = class UserManager {
         }
     }
 
+    static getIsOnlyAdminAllowed(userId) {
+        if (UserManager.userExists(userId)) {
+            return users[userId].getIsOnlyAdminAllowed();
+        }
+    }
```
Comment (arthurshir) [inline userManager.js:161]: Nit: I think this should return a boolean to ensure it adheres to contract of boolean return type

File: botfunctions/isadminmiddleware.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                next();
+            }
+            else {
+                // TODO: tell user only admin is allowed and that they need to configure this in settings
+                return ctx.reply("Dumbass you need to tell the admin to allow everyone to set reminders");
+            }
+        });
+    }
+    next();
+}
```
[the remark this answers is not in the document]
Comment (bubakazouba) [reply botfunctions/isadminmiddleware.js:44]: oh yeah dw will update the text and mention the /settings command

File: botfunctions/isadminmiddleware.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
+const UserManager = require("../userManager.js"),
+    bot = require('../bot.js'),
+    tutorial = require("../tutorial/tutorial.js"),
+    tutorialConstants = require("../tutorial/constants.js");
+
+// Returns true if it's not a group chat
+function isUserAdmin(ctx) {
+    return new Promise(resolve => {
+        return bot.telegram.getChatAdministrators(ctx.chat.id).then(admins => {
+            let isAdmin = admins.filter(admin => admin.user.id == ctx.update.message.from.id).length != 0;
```
[the remark this answers is not in the document]
Comment (bubakazouba) [reply botfunctions/isadminmiddleware.js:10]: nice there's probably a use case for this in a few other places of the repo

File: botfunctions/isadminmiddleware.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (arthurshir) [reply botfunctions/isadminmiddleware.js:49]: What about
```
isUpsertReminderEvent(ctx).then(isUpsertReminderEvent => {
    if (!isUpsertReminderEvent) {
        return next();
    }
    return isUserAdmin(ctx).then(isAdmin => {
        // TODO: tell user only admin is allowed and that they need to configure this in settings
        return isAdmin ? next() : ctx.reply("Dumbass you need to tell the admin to allow everyone to set reminders");
    });
});
```

File: botfunctions/isadminmiddleware.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+function isUpsertReminderEvent(ctx) {
+    // TODO update to support non text messages as well
+    return tutorial.identifyEventFromMessage(ctx.message).then(event => {
+        if ([tutorialConstants.EVENTS.LIST_COMMAND, tutorialConstants.EVENTS.HELP_COMMAND].includes(event)) {
+            return false;
+        }
+        else {
+            return true;
+        }
```
[the remark this answers is not in the document]
Comment (arthurshir) [reply botfunctions/isadminmiddleware.js:27]: This code is actually correct, I was unfamiliar with promises. Returning in a then results in the overarching returned promise resolving the returned values.

File: userManager.js (hunk 1 above)
Comment (bubakazouba) [reply userManager.js:161]: The only problem with this is that now we have 2 different places where we set the default value of `isOnlyAdminAllowed`, I have been trying to make UserManager a blind reader of the user object and free it of logic as much as possible. Hopefully this should go away with https://github.com/bubakazouba/remindmebot/issues/55

Comment (bubakazouba) [reply userManager.js:161]: This should be ok for now especially that it preserved the consistency of not returning anything (i.e returning undefined if the user doesn't exist.

**Review by arthurshir — APPROVED** [review]

**Review by arthurshir — APPROVED** [review]

**Pull request merged**